pkg/report/sarif.go

```diff
@@ -259,6 +282,24 @@ func ToPathUri(input string) string {
 	return strings.ReplaceAll(input, "\\", "/")
 }
 
+func getLocations(name, version, path string, foundLocations map[string][]location, pkgs []ftypes.Package) []location {
```

Comment:
In my humble opinion, It's not a clean code, because you changes an argument. it looks like you hide a return value.

honestly, I don't have a good idea how to improve it.

I'd store this location cache inside `SarifWriter` struct and then `getLocations` will be a method of this one:
```go
type SarifWriter struct {
	Output  io.Writer
	Version string
	run     *sarif.Run
        locationCache map[string][]location
}
func (sw *SarifWriter)  getLocations(name, version, path string, pkgs []ftypes.Package) []location {
...
}
```
maybe I'm wrong. wdyt? 
it's just a suggestion

Reply:
good catch!
It is really more understandable.

I changed it.